src/thresholds.service.js

```diff
@@ -103,44 +28,51 @@ class ThresholdsService {
       return
     }
 
-    const { weeklyLevels, weeksOfStock, targetPopulations } = getFactors(stockCount, location)
+    const transformedLocation = transform(location, productCoefficients, stockCount.date)
 
-    if (!(weeklyLevels && weeksOfStock)) {
+    if (!transformedLocation) {
       return
     }
 
-    return Object.keys(weeklyLevels).reduce((index, productId) => {
-      index[productId] = Object.keys(weeksOfStock).reduce((productThresholds, threshold) => {
-        const level = weeklyLevels[productId] * weeksOfStock[threshold]
-        const product = find(products, isId.bind(null, productId))
+    const { weeksOfStock, weeklyLevels, monthlyTargetPopulations } = transformedLocation
 
-        // Default rounding used in VSPMD and highest possible presentation
-        let presentation = 20
+    return products.reduce((index, product) => {
+      const productId = product._id
+      const weeklyLevel = weeklyLevels[productId]
 
-        if (product && product.presentation) {
-          // TODO: product presentations should be ints, not strings
-          presentation = parseInt(product.presentation, 10)
-        }
+      // Default rounding used in VSPMD and highest possible presentation
+      let presentation = 20
+
+      if (product && product.presentation) {
+        // TODO: product presentations should be ints, not strings
+        presentation = parseInt(product.presentation, 10)
+      }
 
+      index[productId] = Object.keys(weeksOfStock).reduce((productThresholds, threshold) => {
+        const level = weeklyLevel * weeksOfStock[threshold]
         const roundedLevel = Math.ceil(level / presentation) * presentation
         productThresholds[threshold] = roundedLevel
 
-        if (location.level === 'zone' && requiredStateStoresAllocation[productId]) {
+        if (transformedLocation.level === 'zone' && requiredStateStoresAllocation[productId]) {
```

Comment:
Why does `transform` duplicate the `level`? Can't the one on `location` be used?